dlib/image_transforms/interpolation_abstract.h

```diff
@@ -436,6 +436,86 @@ namespace dlib
             - Returns immediately, if size_scale == 1.0
     !*/
 
+// ----------------------------------------------------------------------------------------
+
+    template <
+        typename image_type1,
+        typename image_type2,
+        typename interpolation_type
+        >
+    point_transform_affine letterbox_image (
+        const image_type1& img_in,
+        image_type2& img_out,
+        long size
+        const interpolation_type interp
+    );
+    /*!
+        requires
+            - image_type1 == an image object that implements the interface defined in
+              dlib/image_processing/generic_image.h
+            - image_type2 == an image object that implements the interface defined in
+              dlib/image_processing/generic_image.h
+            - interpolation_type == interpolate_nearest_neighbor, interpolate_bilinear,
+              interpolate_quadratic, or a type with a compatible interface.
+            - size > 0
+            - is_same_object(in_img, out_img) == false
+        ensures
+            - scales in_img so that it fits into a size * size square.
```

Comment:
Add a line that specifically says img_out.nr() == size and img_out.nc() == size.  Otherwise it's not totally obvious that this is so.

Reply:
Same for the other functions below too.  Like be very explicit about the output image sizes.

Reply:
done :)